[bug] profile include() does not work with os=Linux

### Describe the bug

(conan 2.4.0, on linux)


I have a base profile, `base-linux`:
```
[settings]
arch=x86_64
os=Linux
```

I have another profile, `base-clang`:
```
include(base-linux)
[settings]
compiler=clang
compiler.libcxx=libstdc++11
compiler.version=18
compiler.cppstd=20
```

And I have the top level profile, `reldeb-clang`:
```
include(base-clang-18)

[settings]
build_type=RelWithDebInfo
```

When I build with this as the host profile, the resulting output is:
```
======== Input profiles ========
Profile host:
[settings]
build_type=RelWithDebInfo
compiler=clang
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=18
```

Note, there is no os (and arch) listed, and the recipes complain and fail to build.

If I move the base-linux contents into base-clang, then it works.

There seems to be no documentation on the website about this behaviour (that I could see),
and seems to be a bug.
If the user can't do this, then conan should detect this problem and fail earlier with a clear error message.


### How to reproduce it

_No response_

Ok thanks, there was a problem with my test,
1) I have an empty default profile, so I needed to specify --profile:build
2) My actual profile has a second [settings] section, which is the real problem.

I'm aware that ini file formats have limitations, but having a second group results in the first group's settings ignored.

ie I've changed the test, the profile now looks like:
```
    [settings]
    arch=x86_64
    os=Linux
    build_type=Release
    compiler=clang
    compiler.libcxx=libstdc++11
    compiler.version=18
    compiler.cppstd=20

    [conf]
    tools.build:compiler_executables={'c': '/usr/bin/clang-18', 'cpp': '/usr/bin/clang++-18'}

    [settings]
    package/*:option=Whatever
```

Note that second [settings] group
The output is now:
```
E           conans.errors.ConanInvalidConfiguration: conanfile.py (hello/1.0): Invalid ID: Invalid: 'settings.os' value not defined
```

While I agree this is user-error, profile files can be long, and the cause is not obvious.
I think conan's handling of the profile files should change to either:
a) consider ALL items in all matching [group] sections
or, b) error if there is a repeated [group]

ie the user won't be aware that some of the profile settings will be completely ignored.

I've pushed an update to the test, not sure how to append to your PR, I'll mention it there

--- a/conans/client/profile_loader.py
+++ b/conans/client/profile_loader.py
@@ -263,10 +263,6 @@ class _ProfileValueParser(object):
                                                          "options", "conf", "buildenv", "runenv",
                                                          "replace_requires", "replace_tool_requires",
                                                          "runner"])
-        if doc.duplicated_sections:
-            sections = ', '.join(sorted([f'[{s}]' for s in doc.duplicated_sections]))
-            ConanOutput().warning(f"Profile file has duplicated sections: {sections}. It could cause"
-                                  f" some unexpected behavior.")
         # Parse doc sections into Conan model, Settings, Options, etc
         settings, package_settings = _ProfileValueParser._parse_settings(doc)
         options = Options.loads(doc.options) if doc.options else None

--- a/conans/client/profile_loader.py
+++ b/conans/client/profile_loader.py
@@ -263,7 +263,10 @@ class _ProfileValueParser(object):
                                                          "options", "conf", "buildenv", "runenv",
                                                          "replace_requires", "replace_tool_requires",
                                                          "runner"])
-
+        if doc.duplicated_sections:
+            sections = ', '.join(sorted([f'[{s}]' for s in doc.duplicated_sections]))
+            ConanOutput().warning(f"Profile file has duplicated sections: {sections}. It could cause"
+                                  f" some unexpected behavior.")
         # Parse doc sections into Conan model, Settings, Options, etc
         settings, package_settings = _ProfileValueParser._parse_settings(doc)
         options = Options.loads(doc.options) if doc.options else None

--- a/conans/util/config_parser.py
+++ b/conans/util/config_parser.py
@@ -29,7 +29,6 @@ class ConfigParser(object):
     """
     def __init__(self, text, allowed_fields=None, strip_comments=False):
         self._sections = {}
-        self._duplicated_sections = set()
         self._allowed_fields = allowed_fields or []
         pattern = re.compile(r"^\[([a-z_]{2,50})]")
         current_lines = None
@@ -45,9 +44,10 @@ class ConfigParser(object):
                 if self._allowed_fields and field not in self._allowed_fields:
                     raise ConanException("ConfigParser: Unrecognized field '%s'" % field)
                 current_lines = []
+                # Duplicated section
                 if field in self._sections:
-                    # WARNING: This section is already defined. It'll override the existing one.
-                    self._duplicated_sections.add(field)
+                    raise ConanException(f"ConfigParser: Duplicated section: [{field}]. It could"
+                                         f"cause some unexpected behavior.")
                 self._sections[field] = current_lines
             else:
                 if current_lines is None:
@@ -57,10 +57,6 @@ class ConfigParser(object):
                     line = line.split('    #', 1)[0]
                     line = line.strip()
                 current_lines.append(line)
-
-    @property
-    def duplicated_sections(self):
-        return self._duplicated_sections
 
     def line_items(self):
         # Used atm by load_binary_info()

--- a/conans/util/config_parser.py
+++ b/conans/util/config_parser.py
@@ -46,8 +46,7 @@ class ConfigParser(object):
                 current_lines = []
                 # Duplicated section
                 if field in self._sections:
-                    raise ConanException(f"ConfigParser: Duplicated section: [{field}]. It could"
-                                         f"cause some unexpected behavior.")
+                    raise ConanException(f"ConfigParser: Duplicated section: [{field}]")
                 self._sections[field] = current_lines
             else:
                 if current_lines is None:

--- a/conans/util/config_parser.py
+++ b/conans/util/config_parser.py
@@ -29,6 +29,7 @@ class ConfigParser(object):
     """
     def __init__(self, text, allowed_fields=None, strip_comments=False):
         self._sections = {}
+        self._duplicated_sections = set()
         self._allowed_fields = allowed_fields or []
         pattern = re.compile(r"^\[([a-z_]{2,50})]")
         current_lines = None
@@ -44,6 +45,9 @@ class ConfigParser(object):
                 if self._allowed_fields and field not in self._allowed_fields:
                     raise ConanException("ConfigParser: Unrecognized field '%s'" % field)
                 current_lines = []
+                if field in self._sections:
+                    # WARNING: This section is already defined. It'll override the existing one.
+                    self._duplicated_sections.add(field)
                 self._sections[field] = current_lines
             else:
                 if current_lines is None:
@@ -53,6 +57,10 @@ class ConfigParser(object):
                     line = line.split('    #', 1)[0]
                     line = line.strip()
                 current_lines.append(line)
+
+    @property
+    def duplicated_sections(self):
+        return self._duplicated_sections
 
     def line_items(self):
         # Used atm by load_binary_info()

--- a/test/integration/configuration/profile_test.py
+++ b/test/integration/configuration/profile_test.py
@@ -611,5 +611,5 @@ def test_consumer_invalid_profile_multiple_groups():
     })
     client.run("build . --profile:host myprofs/myprofile -g CMakeToolchain",
                assert_error=True)
-    assert "Profile file has duplicated sections: [conf], [settings]" in client.out
-    assert "Invalid: 'settings.os' value not defined" in client.out
+    assert ("ERROR: Error reading 'myprofs/myprofile' profile: ConfigParser: "
+            "Duplicated section: [settings]") in client.out

--- a/test/integration/configuration/profile_test.py
+++ b/test/integration/configuration/profile_test.py
@@ -574,3 +574,48 @@ def test_consumer_specific_settings_from_profile():
     client.save({"conanfile.py": conanfile, "my_profile.txt": profile})
     client.run("install . --profile my_profile.txt")
     assert "I'm hello and my build type is Debug" in client.out
+
+
+def test_consumer_several_local_profiles():
+    """
+    Issue related: https://github.com/conan-io/conan/issues/16448
+    """
+    client = TestClient()
+    conanfile = GenConanfile(name="hello", version="1.0").with_settings("os", "arch",
+                                                                        "build_type", "compiler")
+    prof1 = textwrap.dedent("""\
+    include(base-linux)
+
+    [settings]
+    # arch=x86_64
+    # os=Linux
+    compiler=clang
+    compiler.libcxx=libstdc++11
+    compiler.version=18
+    compiler.cppstd=20
+
+    [conf]
+    tools.build:compiler_executables={'c': '/usr/bin/clang-18', 'cpp': '/usr/bin/clang++-18'}
+    """)
+    prof2 = textwrap.dedent("""\
+    [settings]
+    arch=x86_64
+    os=Linux
+    """)
+    prof3 = textwrap.dedent("""\
+    include(base-clang-18)
+
+    [settings]
+    build_type=Release
+    """)
+    client.save({
+        "conanfile.py": conanfile,
+        "myprofs/base-clang-18": prof1,
+        "myprofs/base-linux": prof2,
+        "myprofs/release-clang-18": prof3,
+    })
+    client.run("build . --profile:host myprofs/release-clang-18 -g CMakeToolchain")
+    # No issues here, it gets the build-step which is empty
+    assert "Invalid: 'settings.os' value not defined" not in client.out
+    assert "Invalid: 'settings.arch' value not defined" not in client.out
+    assert "======== Calling build() ========" in client.out

--- a/test/integration/configuration/profile_test.py
+++ b/test/integration/configuration/profile_test.py
@@ -576,7 +576,7 @@ def test_consumer_specific_settings_from_profile():
     assert "I'm hello and my build type is Debug" in client.out
 
 
-def test_consumer_several_local_profiles():
+def test_consumer_invalid_profile_multiple_groups():
     """
     Issue related: https://github.com/conan-io/conan/issues/16448
     """
@@ -584,11 +584,10 @@ def test_consumer_several_local_profiles():
     conanfile = GenConanfile(name="hello", version="1.0").with_settings("os", "arch",
                                                                         "build_type", "compiler")
     prof1 = textwrap.dedent("""\
-    include(base-linux)
-
     [settings]
-    # arch=x86_64
-    # os=Linux
+    arch=x86_64
+    os=Linux
+    build_type=Release
     compiler=clang
     compiler.libcxx=libstdc++11
     compiler.version=18
@@ -596,26 +595,21 @@ def test_consumer_several_local_profiles():
 
     [conf]
     tools.build:compiler_executables={'c': '/usr/bin/clang-18', 'cpp': '/usr/bin/clang++-18'}
-    """)
-    prof2 = textwrap.dedent("""\
-    [settings]
-    arch=x86_64
-    os=Linux
-    """)
-    prof3 = textwrap.dedent("""\
-    include(base-clang-18)
 
     [settings]
-    build_type=Release
+    # Empty and duplicated section
+
+    [options]
+    package/*:option=Whatever
+
+    [conf]
+    # Another one
     """)
     client.save({
         "conanfile.py": conanfile,
-        "myprofs/base-clang-18": prof1,
-        "myprofs/base-linux": prof2,
-        "myprofs/release-clang-18": prof3,
+        "myprofs/myprofile": prof1
     })
-    client.run("build . --profile:host myprofs/release-clang-18 -g CMakeToolchain")
-    # No issues here, it gets the build-step which is empty
-    assert "Invalid: 'settings.os' value not defined" not in client.out
-    assert "Invalid: 'settings.arch' value not defined" not in client.out
-    assert "======== Calling build() ========" in client.out
+    client.run("build . --profile:host myprofs/myprofile -g CMakeToolchain",
+               assert_error=True)
+    assert "Profile file has duplicated sections: [conf], [settings]" in client.out
+    assert "Invalid: 'settings.os' value not defined" in client.out

